pkg/cloudprovider/provider/vsphere/helper.go

```diff
@@ -78,20 +75,6 @@ func createClonedVM(ctx context.Context, vmName string, config *Config, dc *obje
 		targetVMFolder = datacenterFolders.VmFolder
 	}
 
-	// Create snapshot of the template VM if not already snapshotted.
-	snapshot, err := findSnapshot(ctx, templateVM, snapshotName)
```

Comment:
For some reason I don't know the disk resizing silently doesn't happen when cloning a snapshot.

Since the only advantage of the snapshot cloning is that theoretically the templaveVM could be powered on, its IMHO fine to get rid of all that code and directly clone the templateVM